projects/103_hmwrk/src/main.c

```diff
@@ -0,0 +1,55 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress adc_addr = {
+  .port = GPIO_PORT_A,
+  .pin = 6,
+};
+
+static GpioAddress button_addr = {
```

Comment:
Nit: all static variables should be prefixed with `s_`. Alternatively, the button and interrupts could be declared in main, since this is the only place you use them.

Also, since they won't be changed it is good practice to declare them as `const`.